robottelo/cli/base.py

```diff
@@ -193,7 +195,18 @@ def create(cls, options=None, timeout=None):
                     raise CLIError(tmpl.format(cls.__name__))
                 info_options['organization-id'] = options['organization-id']
 
-            new_obj = cls.info(info_options)
+            # organization creation can take some time
+            if cls.command_base == 'organization':
+                new_obj = wait_for(
+                    lambda: cls.info(info_options),
+                    timeout=300,
+                    delay=15,
```

Comment:
Isn't that a little too extreme? I would expect "some time" to take 2-3 seconds, not 15. Maybe change that to `5`?

Also, isn't nailgun synchronous? And shouldn't Satellite return data only after it is stored and can be used? Kind of looks like workaround for bug in product, might be worth having discussion with devs.

Reply:
I'd like to know the resolution of that conversation.

Reply:
Well things aren't that dreary on real-life Satellite, it's just that having hundreds of orgs in place makes things slower. I don't see this as a bug, I mean if the org is not ready, why should it appear in hammer info? Should hammer info wait until org is created? What if the task gets stuck? Filing would imho just feed backlog

Reply:
> I mean if the org is not ready, why should it appear in hammer info?

I see it rather as: If the org is not ready, why did `hammer create` finish and return its id, and why this id can't be used immediately?

Either way, that was deep in suggestion and speculation territory. I'm OK with not following up with dev.

Reply:
Mirek, that is a good point and a good potential bug if things truly aren't ready when create exists.